gcp-observability/src/main/java/io/grpc/gcp/observability/ObservabilityConfigImpl.java

```diff
@@ -50,7 +50,7 @@ final class ObservabilityConfigImpl implements ObservabilityConfig {
   private static final double EPSILON = 1e-6;
 
   private static final Pattern METHOD_NAME_REGEX =
-      Pattern.compile("^([*])|((([\\w]+)/((?:\\w+)|[*])))$");
+      Pattern.compile("^([*])|((([\\w.]+)/((?:\\w+)|[*])))$");
```

Comment:
I think this regex also allows strings like `"package."` i.e. a trailing "." or even a leading dot like `".service"` which are technically invalid names. Will it be quick to fix it properly? Or should we plan on a TODO to be fixed after this release?